packages/@aws-cdk/aws-pipes-sources-alpha/lib/dynamodb.ts

```diff
@@ -0,0 +1,187 @@
+import { IPipe, ISource, SourceConfig } from '@aws-cdk/aws-pipes-alpha';
+import { Duration } from 'aws-cdk-lib';
+import { ITableV2 } from 'aws-cdk-lib/aws-dynamodb';
+import { IRole } from 'aws-cdk-lib/aws-iam';
+import { ITopic, Topic } from 'aws-cdk-lib/aws-sns';
+import { IQueue, Queue } from 'aws-cdk-lib/aws-sqs';
+import { DynamoDBStartingPosition, OnPartialBatchItemFailure } from './enums';
+
+/**
+ * Parameters for the DynamoDB source.
+ */
+export interface DynamoDBSourceParameters {
+  /**
+    * The maximum number of records to include in each batch.
+    *
+    * Minumum: 1
+    * Maxiumum: 10000
+    *
+    * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcedynamodbstreamparameters.html#cfn-pipes-pipe-pipesourcedynamodbstreamparameters-batchsize
+    * @default 1
+    */
+  readonly batchSize?: number;
+
+  /**
+    * Define the target SQS queue or SNS topic to send dead-letter queue events to.
+    *
+    * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcedynamodbstreamparameters.html#cfn-pipes-pipe-pipesourcedynamodbstreamparameters-deadletterconfig
+    * @default - no dead letter queue
+    */
+  readonly deadLetterTarget?: IQueue | ITopic;
+
+  /**
+   * The maximum length of a time to wait for events.
+   *
+   * Minumum: Duration.seconds(0)
+   * Maxiumum: Duration.seconds(300)
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcedynamodbstreamparameters.html#cfn-pipes-pipe-pipesourcedynamodbstreamparameters-maximumbatchingwindowinseconds
+   * @default - no batching window
```

Comment:
It might be better to make this more explicit, something like "the events will be treated immediately"